src/components/molecules/__tests__/TabsContainer.test.tsx

```diff
@@ -0,0 +1,36 @@
+import 'jest-dom/extend-expect';
+import React from 'react';
+import { cleanup, fireEvent, render } from '../../../helpers/test-utils';
+import Tab from '../Tab';
+import TabsContainer from '../TabsContainer';
+
+afterEach(cleanup);
+
+test('TabsContainer renders tabs and tabpanels correctly, and switches tabpanel when a tab is clicked.', () => {
+  const label1 = 'Tab-1';
+  const content1 = 'Tab-1 content';
+  const label2 = 'Tab-2';
+  const content2 = 'Tab-2 content';
+
+  const { getByTestId, getByText, queryByText } = render(
+    <TabsContainer>
+      <Tab label={label1}>
+        <h2>{content1}</h2>
+      </Tab>
+      <Tab label={label2}>
+        <h2>{content2}</h2>
+      </Tab>
+    </TabsContainer>
+  );
+
+  expect(getByTestId('Tab-menu'));
+  expect(getByText(label1));
+  expect(getByText(label2));
+  expect(getByText(content1));
+  expect(queryByText(content2)).toBeNull();
+
+  // Switching tabpanels through a click event
+  fireEvent.click(getByText(label2));
+  expect(getByText(content2));
+  expect(queryByText(content1)).toBeNull();
+});
```

Comment:
I made a choice to only test the TabContainer component as the other components used are tightly connected to it. If we  decide that the others should be tested individually as well, I think that should be an issue for later and not prioritized right now.